Content.Server/Atmos/EntitySystems/GasAnalyzableSystem.cs

```diff
@@ -0,0 +1,74 @@
+using Content.Server.Atmos.Components;
+using Content.Server.NodeContainer.NodeGroups;
+using Content.Server.NodeContainer.Nodes;
+using Content.Server.NodeContainer;
+using Content.Server.Tools;
+using Content.Shared.Examine;
+using Content.Shared.Temperature;
+using Content.Shared.Verbs;
+using JetBrains.Annotations;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Atmos.EntitySystems
+{
+    [UsedImplicitly]
+    public sealed class GasAnalyzableSystem : EntitySystem
+    {
+        [Dependency] private readonly ToolSystem _toolSystem = default!;
+        [Dependency] private readonly ExamineSystemShared _examineSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<GasAnalyzableComponent, GetVerbsEvent<ExamineVerb>>(OnGetExamineVerbs);
+        }
+
+        private void OnGetExamineVerbs(EntityUid uid, GasAnalyzableComponent component, GetVerbsEvent<ExamineVerb> args)
+        {
+            // Must be in details range to try this.
+            if (_examineSystem.IsInDetailsRange(args.User, args.Target))
+            {
+                var held = args.Using;
+
+                var enabled = held != null;
```

Comment:
AFAICT this allows people to examine pipes if they're holding literally anything? There should either be a `HasComponent<SharedGasAnalyzerComponent>` or a `_toolSystem.HasQuality` check here.

Reply:
Good catch, you're absolutely right. Will fix.